### PR TITLE
Dropdown list for country codes

### LINKED ISSUE #4113: [FEATURE] Improve Nordigen app
**Is your feature request related to a problem? Please describe.**
In the Nordigen app, when choosing a country, pipedream asks for a two letters code. If you enter `US`, it won't work because the Nordigen app works only for [the following countries](https://nordigen.com/en/coverage/).

**Describe the solution you'd like**
Change the [country prop](https://github.com/PipedreamHQ/pipedream/blob/5def469c88a61e3b93ff31ecfc998f4585156cb9/components/nordigen/nordigen.app.mjs#L7) to have a dropdown list of the supported countries.

**Do you have a workaround?**
No

### PR DESCRIPTION
**Pull request by LucBerge**
- Fix #4113

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4449"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/nordigen/nordigen.app.mjs [hunk 1]
```diff
@@ -6,8 +6,134 @@ export default {
   propDefinitions: {
     country_code: {
       type: "string",
-      label: "Enter a country code",
-      description: "[ISO 3166](https://www.iso.org/obp/ui/en/#iso:pub:PUB500001:en) two-character code for the country (eg. `FR`) for one of [these supported countries](https://nordigen.com/en/coverage/).",
+      label: "Country",
+      description: "Country where your bank is located.",
+	  options: [
```
Comment (feyzullah) [inline components/nordigen/nordigen.app.mjs:11]: Can you please move these options to a common file like `nordigen/common/constants.mjs` like in [this app file](https://github.com/PipedreamHQ/pipedream/blob/1e25f764834e448c49d10a513a295650a2a2828f/components/apitemplate_io/app/apitemplate_io.app.ts#L3) and [this constants file](https://github.com/PipedreamHQ/pipedream/blob/master/components/apitemplate_io/common/constants.ts)? This way app file might be too big.

**Review by feyzullah — CHANGES_REQUESTED** [review]
Hi @LucBerge, thank you very much for your contribution. I requested just one change. Can you please check it and can you please merge `master` to run our checks properly?

File: components/nordigen/nordigen.app.mjs (hunk 1 above)
Comment (LucBerge) [reply components/nordigen/nordigen.app.mjs]: Done

**Comment by LucBerge** [comment]
The `package.json` file is missing. Should I create it as well ?

```json
{
  "name": "@pipedream/nordigen",
  "version": "0.0.1",
  "description": "Pipedream Nordigen Components",
  "main": "nordigen.app.mjs",
  "keywords": [
    "pipedream",
    "nordigen"
  ],
  "homepage": "https://pipedream.com/apps/nordigen",
  "author": "Pipedream <support@pipedream.com> (https://pipedream.com/)",
  "license": "MIT",
  "dependencies": {
    "axios": "^0.21.1",
    "js-base64": "^3.7.2",
    "mime-types": "^2.1.35"
  },
  "publishConfig": {
    "access": "public"
  }
}

```

**Comment by feyzullah** [comment]
> 

@LucBerge yes, please. If possible, can you please use [@pipedream/platform axios](https://pipedream.com/docs/pipedream-axios/#pipedream-platform-axios)?

**Comment by LucBerge** [comment]
> @LucBerge yes, please. If possible, can you please use [@pipedream/platform axios](https://pipedream.com/docs/pipedream-axios/#pipedream-platform-axios)?

From what I remember,  [@pipedream/platform axios](https://pipedream.com/docs/pipedream-axios/#pipedream-platform-axios)? is not working for a source.

File: components/nordigen/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "main": "nordigen.app.mjs",
+  "keywords": [
+    "pipedream",
+    "nordigen"
+  ],
+  "homepage": "https://pipedream.com/apps/nordigen",
+  "author": "Pipedream <support@pipedream.com> (https://pipedream.com/)",
+  "license": "MIT",
+  "dependencies": {
+    "axios": "^0.21.1",
```
Comment (LucBerge) [inline components/nordigen/package.json:14]: This is not the correct value. What should I put ?

Comment (feyzullah) [reply components/nordigen/package.json:14]: We can definitely use latest version

**Comment by feyzullah** [comment]
> > @LucBerge yes, please. If possible, can you please use [@pipedream/platform axios](https://pipedream.com/docs/pipedream-axios/#pipedream-platform-axios)?
> 
> From what I remember, we cannot use [@pipedream/platform axios](https://pipedream.com/docs/pipedream-axios/#pipedream-platform-axios) for a source.

It is used in new-transaction and in the app file. It seems that it was done before, so we can ignore that

File: components/nordigen/package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  },
```
Comment (feyzullah) [inline components/nordigen/package.json:15]:
```suggestion
    "axios": "^0.21.1"
  },
```
It seems that there is an extra comma here

**Comment by Hassan-Abbas-SQA** [comment]
@LucBerge TestPassed 
image

**Pull request merged**